tests/main/views/test_feedback.py

```diff
@@ -0,0 +1,38 @@
+from lxml import html
+
+import mock
+from ...helpers import BaseApplicationTest
+
+
+class TestFeedbackForm(BaseApplicationTest):
+    def _post(self):
+        return self.client.post('/feedback', data={
+            'uri': 'test:some-uri',
+            'what_doing': 'test: what doing text',
+            'what_happened': 'test: what happened text'})
+
+    @mock.patch('requests.post')
+    def test_google_gone_gives_503(self, external_requests_post):
+        assert self._post().status_code == 503
+
+    @mock.patch('requests.post')
+    def test_feedback_submission(self, external_requests_post):
+        external_requests_post.return_value.status_code = 200
+        response = self._post()
+
+        assert response.status_code == 303
+        new_page = self.client.get(response.location)
+        assert "Thank you for your message" in new_page.get_data(as_text=True)
+
+    def test_feedback_form_has_csrf(self):
```

Comment:
If this is from the frontend-toolkit, we shouldn't really test for this here, should we?

Reply:
Hmm I wondered if you'd notice that. I might try to argue that where the form happens to come from is an implementation detail. But then you'd probably make me duplicate the test into all the frontend apps.

Ho hum. I'll have a look and see if there's anywhere in the toolkit that this test could meaningfully go, but if not I'd still rather have it here than not at all?

Reply:
Yep, if there's nowhere reasonable for it to go in the toolkit, this is indeed better than nothing. 👍

Reply:
I think it's a good habit to test for things you "expect" from an underlying component. The only reason not to is if it's superfluously laborious. I don't believe in this strict classification of tests into "unit tests" and "integration tests". It's important to check that when using a lib/API, you've actually got the right end of the stick over how things work.

Reply:
(it also will alert you to breakage through api changes)